src/Tring/OutputPrinter.cs

```diff
@@ -16,78 +16,95 @@
-            PrintRequest(status);
-            PrintResult(status.Connect, status.ConnectionTimeMs);
-            PrintPing(status.PingResult, status.PingTimeMs);
-            PrintLocalInterface(status.LocalInterface);
-            PrintProtocol(status.Request.Port);
-            PrintHostName(status.Request.Url);
+            _lockObject = new object();
+            _lenghtIP = lenghtIP >= _minimumLengthIP ? lenghtIP : _minimumLengthIP;
+            _lenghtPort = lenghtPort >= _minimumLengthPort ? lenghtPort : _minimumLengthPort;
         }
 
-        public static void PrintTable()
+        public void PrintLogEntry(DateTimeOffset startTime, ConnectionResult status, int index)
         {
-            Console.WriteLine("| Time              | IP              | Port  | Connect | Ping    | Local Interface | Protocol | Hostname  ");
-            // example output  | 20:22:22-20:23:33 | 100.100.203.104 | 80222 | Timeout | 1000 ms | 111.111.111.111 | https    | google.com
+            lock (_lockObject)
+            {
+                SetPrintLine(index);
+                PrintTime(startTime, status.TimeStamp);
+                PrintRequest(status);
+                PrintResult(status.Connect, status.ConnectionTimeMs);
+                PrintPing(status.PingResult, status.PingTimeMs);
+                PrintLocalInterface(status.LocalInterface);
+                PrintProtocol(status.Request.Port);
+                PrintHostName(status.Request.Url); 
+            }
+        }
+
+        public void PrintTable()
+        {
+            Console.WriteLine(" | Time              | "+ "IP".PadRight(_lenghtIP)+ " | "+"Port".PadRight(_lenghtPort)+" | Connect | Ping    | "+"Egress".PadRight(_lenghtIP)+" | Protocol | Hostname");
+            // example output   | 20:22:22-20:23:33 | 100.100.203.104 | 80222 | Timeout | 1000 ms | 111.111.111.111 | https    | google.com
         }
 
-        public static void ResetPrintLine()
+        public static void SetPrintLine(int lines = 0)
         {
             if (!Console.IsOutputRedirected)
-                Console.SetCursorPosition(0, Console.CursorTop);
+                Console.SetCursorPosition(0, lines);
         }
 
-        public static void HideCursor()
+        public  void HideCursor()
         {
             if (!Console.IsOutputRedirected)
                 Console.CursorVisible = false;
         }
 
-        public static void CleanUp()
+        public static void CleanUp(int endLine)
         {
             if (!Console.IsOutputRedirected)
             {
                 Console.CursorVisible = true;
                 Console.ResetColor();
-                Console.CursorTop++;
+                if (Console.CursorTop <= endLine)
+                    SetPrintLine(endLine+1);
+                else
+                    Console.CursorTop++;
             }
         }
-        public static void SetupCleanUp()
+        public void SetupCleanUp(CancellationTokenSource sourceToken, int endLine)
```

Comment:
I think the name `SetupCleanUp` is a bit vague. Maybe something like `RegisterCancelKeyPressHandler`?